src/views/Opportunities/OpportunityComponents/OpportunityForm.jsx

```diff
@@ -143,6 +144,16 @@ const OpportunityForm = (props) => {
         assigned_to: props?.assigned_to?.id,
         employer: props.employer ? Number(props.employer.id) : '',
       });
+
+      const searchedEmployer = employerOptions.find((option) => option.label === props.employer.name);
+      if (!searchedEmployer) {
+        const newvalue = {
+          key: props.employer.name,
+          label: props.employer.name,
+          value: Number(props.employer.id),
+        };
+        employerOptions.push(newvalue);
+      }
```

Comment:
This needs to be inside the `getAllEmployers.then` callback. You check for the searchedEmployer before setting the employer options for the dropdown

The code should look like this:
```js
getAllEmployers().then(data => {
  const employersData = data?.data?.data?.employers.map((employer) => ({
    key: employer.name,
    label: employer.name,
    value: Number(employer.id),
    details: employer,
  }));
  const searchedEmployer = employersData.find((option) => option.value === Number(props.employer.id));
  if (!searchedEmployer) {
    employersData.push({
      key: props.employer.name,
      label: props.employer.name,
      value: Number(props.employer.id),
    });
  }
  setEmployerOptions(employersData);
});
```